pkg/api/aim2/service/app/service.go

```diff
@@ -0,0 +1,95 @@
+package app
+
+import (
+	"context"
+
+	"github.com/G-Research/fasttrackml/pkg/api/aim2/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/aim2/dao/convertors"
+	"github.com/G-Research/fasttrackml/pkg/api/aim2/dao/repositories"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/pkg/common/api"
+	"github.com/G-Research/fasttrackml/pkg/database"
+)
+
+// Service provides service layer to work with `app` business logic.
+type Service struct {
+	appRepository repositories.AppRepositoryProvider
+}
+
+// NewService creates new Service instance.
+func NewService(appRepository repositories.AppRepositoryProvider) *Service {
+	return &Service{
+		appRepository: appRepository,
+	}
+}
+
+// Get returns app object.
+func (s Service) Get(
+	ctx context.Context, namespace *models.Namespace, req *request.GetAppRequest,
+) (*database.App, error) {
+	app, err := s.appRepository.GetByNamespaceIDAndAppID(ctx, namespace.ID, req.ID.String())
+	if err != nil {
+		return nil, api.NewInternalError("unable to find app by id %q: %s", req.ID, err)
+	}
+	if app == nil {
+		return nil, api.NewResourceDoesNotExistError("app '%s' not found", req.ID)
+	}
+	return app, nil
+}
+
+// Create creates new app object.
+func (s Service) Create(
+	ctx context.Context, namespace *models.Namespace, req *request.CreateAppRequest,
+) (*database.App, error) {
+	app := convertors.ConvertCreateAppRequestToDBModel(namespace, req)
+	if err := s.appRepository.Create(ctx, app); err != nil {
+		return nil, api.NewInternalError("unable to create app: %v", err)
+	}
+	return app, nil
+}
+
+// Update updates existing app object.
+func (s Service) Update(
+	ctx context.Context, namespace *models.Namespace, req *request.UpdateAppRequest,
+) (*database.App, error) {
+	app, err := s.appRepository.GetByNamespaceIDAndAppID(ctx, namespace.ID, req.ID.String())
+	if err != nil {
+		return nil, api.NewInternalError("unable to find app by id %s: %s", req.ID, err)
+	}
+	if app == nil {
+		return nil, api.NewResourceDoesNotExistError("app with id '%s' not found", req.ID)
+	}
+
+	app.Type = req.Type
+	app.State = database.AppState(req.State)
+
+	if err := s.appRepository.Update(ctx, app); err != nil {
+		return nil, api.NewInternalError("unable to update app '%s': %s", app.ID, err)
+	}
+	return app, nil
+}
+
+// GetApps returns the list of active apps.
+func (s Service) GetApps(ctx context.Context, namespace *models.Namespace) ([]database.App, error) {
```

Comment:
I don't think we should return `database` package here. I'd think we reserve `database` module for migrations, and use `api2/dao/models` in the app logic.